app/controllers/concerns/collection.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Generic methods to be used when calling resolve_collection on our models
+module Collection
+  extend ActiveSupport::Concern
+
+  include Sorting
+
+  included do
+    def resolve_collection
+      result = search(sort(policy_scope(resource)))
+      result.paginate(page: pagination_offset, per_page: pagination_limit)
+    end
+
+    private
+
+    def search(data)
+      return data if params[:search].blank?
+
+      data.search_for(params[:search])
+    end
+
+    def sort(data)
+      return data if params[:sort_by].blank?
+
+      by = params[:sort_by].each_with_object({}) do |item, obj|
```

Comment:
This does not handle single param cases gracefully:
```
GET /api/compliance/v1/systems?sort_by=id

undefined method `each_with_object' for "id":String
```

Reply:
Yeah, that's kinda against the specifications, the `sort_by` should be always an array.

Reply:
Can it be handled gracefully please? Instead of a `500 Internal Server Error` return 4XX, and not raise an alarm.

Personally, I'd still handle it even w/o using `[]`. I guess that there is not even a standard for handling an array from query string.

Reply:
:thinking: well, it's handled as any other invalid param, I don't see the point why this should get a special treatment. And yes, arrays in HTTP are a pretty standard thing.

Reply:
It's ok to extend the spec. Ruby idiom for forcing an array: `[maybe_an_array].flatten`

Reply:
My bad, the spec actually has the non-array version :see_no_evil: fixing

Reply:
Done